dynosaur/tests/pass/multiple-lifetimes-impl-trait.stdout

```diff
@@ -0,0 +1,84 @@
+#![feature(prelude_import)]
+#[prelude_import]
+use std::prelude::rust_2021::*;
+#[macro_use]
+extern crate std;
+trait Trait<'a, 'b> {}
+impl<T> Trait<'_, '_> for T {}
+
+trait SomeTrait {
+    async fn async_ret_impl_trait<'a, 'b>(&self, a: &'a u8, b: &'b u8)
+    -> impl Trait<'a, 'b>;
+}
+mod _dynosaur_macro_dynsometrait {
+    use super::*;
+    trait ErasedSomeTrait {
+        fn async_ret_impl_trait<'a, 'b, 'life0,
+        'dynosaur>(&'life0 self, a: &'a u8, b: &'b u8)
+        ->
+            ::core::pin::Pin<Box<dyn ::core::future::Future<Output =
+            impl Trait<'a, 'b>> + 'dynosaur>>
+        where
+        'a: 'dynosaur,
+        'b: 'dynosaur,
+        'life0: 'dynosaur,
+        Self: 'dynosaur;
+    }
+    impl<DYNOSAUR: SomeTrait> ErasedSomeTrait for DYNOSAUR {
+        fn async_ret_impl_trait<'a, 'b, 'life0,
+            'dynosaur>(&'life0 self, a: &'a u8, b: &'b u8)
+            ->
+                ::core::pin::Pin<Box<dyn ::core::future::Future<Output =
+                impl Trait<'a, 'b>> + 'dynosaur>> where 'a: 'dynosaur,
+            'b: 'dynosaur, 'life0: 'dynosaur, Self: 'dynosaur {
+            Box::pin(<Self as SomeTrait>::async_ret_impl_trait(self, a, b))
+        }
+    }
+    #[repr(transparent)]
+    pub struct DynSomeTrait<'dynosaur_struct> {
+        ptr: dyn ErasedSomeTrait + 'dynosaur_struct,
+    }
+    impl<'dynosaur_struct> SomeTrait for DynSomeTrait<'dynosaur_struct> {
+        fn async_ret_impl_trait<'a, 'b>(&self, a: &'a u8, b: &'b u8)
+            -> impl ::core::future::Future<Output = impl Trait<'a, 'b>> {
```

Comment:
This doesn't actually compile. Seems like async fn with RPITIT is not yet supported (fine).

I would like it if we built all tests in this directory.